webplugin/js/app/mck-sidebox-1.0.js

```diff
@@ -9718,6 +9717,41 @@ var userOverride = {
                 }
                 return '';
             };
+            _this.getMessageCreatedAtTime = function (createdAtTime) {
+                if (TIME_FORMAT_24_HOURS) {
+                    var msgtime = new Date(createdAtTime);
+                    var yesterday = new Date();
+                    if (yesterday.getDate() != msgtime.getDate()) {
+                        var time = msgtime.toLocaleString('en-US', {
+                            day: 'numeric',
+                            month: 'short'
+                        }) + ", " + msgtime.getHours() + ":" + msgtime.getMinutes();
+                        return time;
+                    }
+                    else if (yesterday.getDate() == msgtime.getDate() && yesterday.getMonth() != msgtime.getMonth()) {
+                        var time = msgtime.toLocaleString('en-US', {
+                            day: 'numeric',
+                            month: 'short'
+                        }) + ", " + msgtime.getHours() + ":" + msgtime.getMinutes();
+                        return time;
+                    }
+                    else if (yesterday.getDate() == msgtime.getDate() && yesterday.getMonth() == msgtime.getMonth() && yesterday.getFullYear() != msgtime.getFullYear()) {
+                        var time = msgtime.toLocaleString('en-US', {
```

Comment:
is this code same as above else if?
if yes then why do we need multiple else if if the code is going to be same in both

Reply:
there will be instance where month will also be same like after 1 year of message
so we will need to check for year